source/imaer-util/src/test/java/nl/overheid/aerius/validation/ManureStorageValidatorTest.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Crown copyright
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+package nl.overheid.aerius.validation;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import nl.overheid.aerius.shared.domain.v2.source.ManureStorageEmissionSource;
+import nl.overheid.aerius.shared.domain.v2.source.manure.CustomManureStorage;
+import nl.overheid.aerius.shared.domain.v2.source.manure.StandardManureStorage;
+import nl.overheid.aerius.shared.emissions.FarmEmissionFactorType;
+import nl.overheid.aerius.shared.exception.AeriusException;
+import nl.overheid.aerius.shared.exception.ImaerExceptionReason;
+
+@ExtendWith(MockitoExtension.class)
+class ManureStorageValidatorTest {
+
+  private static final String SOURCE_ID = "OurSourceId";
+
+  @Mock ManureStorageValidationHelper validationHelper;
+
+  @Test
+  void testValidCustom() {
+    final ManureStorageEmissionSource source = constructSource();
+    final CustomManureStorage subSource = new CustomManureStorage();
+    subSource.setFarmEmissionFactorType(FarmEmissionFactorType.PER_METERS_SQUARED_PER_DAY);
+    subSource.setMetersSquared(3.2);
+    subSource.setNumberOfDays(40);
+    source.getSubSources().add(subSource);
+
+    final List<AeriusException> errors = new ArrayList<>();
+    final List<AeriusException> warnings = new ArrayList<>();
+    final ManureStorageValidator validator = new ManureStorageValidator(errors, warnings, validationHelper);
+
+    final boolean valid = validator.validate(source);
+
+    assertTrue(valid, "Valid test case");
+    assertTrue(errors.isEmpty(), "No errors");
+    assertTrue(warnings.isEmpty(), "No warnings");
```

Comment:
You could see if you can put the validation/assert into a method with parameters to reduce duplication of code.

Reply:
Good point, see latest commit.